doc/source/dev/index.rst

```diff
@@ -2,15 +2,323 @@
+     before merging. Approval means the core team member has carefully reviewed
+     the changes, and the PR is ready for merging.
+
+5. Document changes
+
+   If your change introduces any user-facing modifications, please update the
+   current release notice under ``doc/release/X.XX-notes.rst``
+
+   If your change introduces a deprecation, make sure to follow the
+   `deprecation policy <http://www.numpy.org/neps/nep-0023-backwards-
+    compatibility.html>`_.
+
+.. note::
+
+   To reviewers: if it is not obvious from the PR description, add a short
+   explanation of what a branch did to the merge message and, if closing a
+   bug, also add "Closes #123" where 123 is the issue number.
+
+
+Divergence between ``upstream master`` and your feature branch
+--------------------------------------------------------------
+
+If GitHub indicates that the branch of your Pull Request can no longer
+be merged automatically, merge the master branch into yours, you must
+resolve the conflict. One way to do this is to `rebase on master <rebasing-on-
+master>`.
+
+Guidelines
+----------
+
+* All code should have tests (see `test coverage`_ below for more details).
+* All code should be `documented <docstring-standard>`.
+* No changes are ever committed without review and approval by two core
+  team members.  Ask on the
+  `mailing list <https://mail.python.org/mailman/listinfo/numpy-discussion>`_ if
+  you get no response to your pull request.
+  **Never merge your own pull request.**
+
+Stylistic Guidelines
+--------------------
+
+* Set up your editor to remove trailing whitespace.  Follow `PEP08
+  <https://www.python.org/dev/peps/pep-0008/>`__.  Check code with pyflakes / flake8.
+
+* Use numpy data types instead of strings (``np.uint8`` instead of
+  ``"uint8"``).
+
+* Use the following import conventions::
+
+   import numpy as np
+   import matplotlib.pyplot as plt
+
+   cimport numpy as cnp  # in Cython code
+
+* Use ``np_intp`` as data type for all indexing, shape and size variables
+  in C/C++ and Cython code.
+
+* Use relative module imports, i.e. ``from .utils import xyz`` rather than
+  ``from numpy.utils import xyz``.
+
+* Wrap Cython code in a pure Python function, which defines the API. This
+  improves compatibility with code introspection tools, which are often not
+  aware of Cython code.
+
+* For Cython functions, release the GIL whenever possible, using
+  ``with nogil:``.
+
+
+Test coverage
+-------------
+
+Pull requests (PRs) that modify code should either have new tests, or modify existing
+tests to fail before the PR and pass afterwards. You should `run the tests
+<development-environment>` before pushing a PR.
+
+Tests for a module should ideally cover all code in that module,
+i.e., statement coverage should be at 100%.
+
+To measure the test coverage, install
+`pytest-cov <https://pytest-cov.readthedocs.io/en/latest/>`__
+(using ``easy_install pytest-cov``) and then run::
```

Comment:
Is this up to date? Wouldn't one use `pip install` these days? (That does work for me)

Reply:
removing, the link has installation instructions